src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,14 +1,43 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
 /**
  * <p>Дано файл, потрібно прочитати його вміст і вибрати всі слова що починаються на `w`.
  * Результат повернути у вигляді відсортованого масиву (за замовчуванням). Всі слова повинні
  * бути в нижньому регістрі. У випадку якщо таких слів не знайдено повернути пустий масив.
  * Приклад: Width world Wide web
  * Результат: web wide width world</p>
  */
+
 public class FileWork {
+    private static final String LETTER = "w";
+    private static final String EMPTY = "";
+    private static final String SPACE = " ";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        String tempString = EMPTY;
+        String[] result;
+        try (BufferedReader newReader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = newReader.readLine()) != null) {
+                String[] arr1 = line.toLowerCase().split(SPACE);
+                for (String line2 : arr1) {
+                    if (line2.startsWith(LETTER)) {
+                        tempString += line2.replaceAll("[^a-z0-9]", "") + SPACE;
```

Comment:
let's use String builder here. Tell me why this would be a better solution in comment below

Reply:
done. 
StringBuilder have more economy than String. 
and can expand